pkg/info/info.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package info
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/log"
+	"github.com/thanos-io/thanos/pkg/info/infopb"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+	"google.golang.org/grpc"
+)
+
+type InfoServer struct {
+	infopb.UnimplementedInfoServer
+
+	logger    log.Logger
+	component infopb.ComponentType
+
+	getLabelSet        func() []labelpb.ZLabelSet
+	getStoreInfo       func() *infopb.StoreInfo
+	getExemplarsInfo   func() *infopb.ExemplarsInfo
+	ruleAPIPresent     bool
+	targetAPIPresent   bool
+	metadataAPIPresent bool
+}
+
+func NewInfoServer(
+	component infopb.ComponentType,
+	getLabelSet func() []labelpb.ZLabelSet,
+	getStoreInfo func() *infopb.StoreInfo,
+	getExemplarsInfo func() *infopb.ExemplarsInfo,
+	ruleAPIPresent bool,
+	targetAPIPresent bool,
+	metadataAPIPresent bool,
+) *InfoServer {
+	return &InfoServer{
+		component:          component,
+		getLabelSet:        getLabelSet,
+		getStoreInfo:       getStoreInfo,
```

Comment:
For convenience, let's allow users to pass a `nil` func. If the func for any of these parameters is `nil`, we replace it with the corresponding `func() *infopb.XInfo { return nil }`